docs/how-to/integrate-external-libraries.md

```diff
@@ -0,0 +1,312 @@
+If the library is only needed by a single component, it is also acceptable to place the `add_subdirectory` code directly in that component's `CMakeLists.txt` file. If the library is needed at the project level, it is best to place the `add_subdirectory` code in the root `project.cmake` file. The bottom line is the following: the `add_subdirectory` command must be called before registering any module that uses the library.
+
+#### Step 2: Set the library as a dependency of the module
+
+Any component or module that depends on ETL can now link against it by adding the following entry to its `MOD_DEPS`:
+
+```cmake
+# In MyComponent/CMakeLists.txt 
+set(SOURCE_FILES
+  "${CMAKE_CURRENT_LIST_DIR}/MyComponent.fpp"
+  "${CMAKE_CURRENT_LIST_DIR}/MyComponent.cpp"
+)
+set(MOD_DEPS
+  etl::etl   # This is the target name for the ETL library
+)
+register_fprime_module()
+```
+
+---
+
+<a id="fetchcontent"></a>
+
+## Approach 1 (bis): FetchContent
+
+|     |     |
+| --- | --- |
+| **Benefits** | Very easy to set up; portable |
+| **Drawbacks** | |
+| **Considerations** | Recommended approach if available |
+
+This method is very similar to the previous one, only it applies if the source code for the library is not available in your source tree, but rather at some remote location (git repository, web archive, etc.). In this case, you can use the [FetchContent module](https://cmake.org/cmake/help/v3.24/module/FetchContent.html) to download the library source code and add it to your project. In many cases, this approach will perform a clone and then a `add_subdirectory()` call ([reference](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable)). For this reason, the same considerations and details apply. 
+
+#### Step 1: Use FetchContent to include the library in your project
+
+The following excerpt demonstrates how to integrate the [ETL libraries](https://github.com/ETLCPP/etl) using `FetchContent`:
+
+```cmake
+# This code can be located in the root project.cmake file if the library is used by multiple modules
+# or directly in the module's CMakeLists.txt if it is only used by that module.
+include(FetchContent)
+FetchContent_Declare(
+    etl
+    GIT_REPOSITORY https://github.com/ETLCPP/etl
+    GIT_TAG        20.40.0
+)
+FetchContent_MakeAvailable(etl)
+```
+
+The same considerations as [Approach 1](#add_subdirectory) apply with regards to where to place that code.
+
+#### Step 2: Set the library as a dependency of the module
+
+Any component or module that depends on ETL can now link against it by adding the following entry to its `MOD_DEPS`:
+
+```cmake
+# In MyComponent/CMakeLists.txt 
+set(SOURCE_FILES
+  "${CMAKE_CURRENT_LIST_DIR}/MyComponent.fpp"
+  "${CMAKE_CURRENT_LIST_DIR}/MyComponent.cpp"
+)
+set(MOD_DEPS
+  etl::etl   # This is the target name for the ETL library
```

Comment:
Should we rework this to use the "new" cmake format?

Reply:
The example isn't using that format yet so I'd rather stick with what's in the example...